cogs/member_stats.py

```diff
@@ -1,15 +1,141 @@
+        if not await is_process_and_user_clear(self.bot, 'm_stats_start', ctx):
+            return
+        self.m_stats.start(ctx)
+        self.isRecording = True
+        await ctx.send(f'statistics recording started')
+
+    @commands.command()
+    async def m_stats_stop(self, ctx):
+        if not await is_moderator(ctx):
+            return
+        if not await is_mod_commands_channel(ctx):
+            return
+        author_id = ctx.author.id
+        if not await is_process_and_user_clear(self.bot, 'm_stats_stop', ctx):
+            return
+        self.m_stats.cancel()
+        self.isRecording = False
+        await ctx.send(f'statistics recording stopped')
+
+    @commands.command()
+    async def m_stats_status(self, ctx):
+        if not await is_moderator(ctx):
+            return
+        if not await is_mod_commands_channel(ctx):
+            return
+
+        author_id = ctx.author.id
+        if not await is_process_and_user_clear(self.bot, 'm_stats_status',
+                                               ctx):
+            return
+        if self.isRecording is True:
+            await ctx.send(f'statistics recording is running')
+        else:
+            await ctx.send(f'statistics recording NOT running')
 
     @commands.command()
+    async def m_stats_show(self, ctx):
+        if not await is_moderator(ctx):
+            return
+        if not await is_mod_commands_channel(ctx):
+            return
+
+        author_id = ctx.author.id
+        if not await is_process_and_user_clear(self.bot, 'm_stats_show', ctx):
+            return
+        if self.isRecording is True:
+            await ctx.send(f'first stop statistics recording ($m_stats_stop)')
+        else:
+            if path.exists('./server_specific/member_stats.csv'):
+                with open('./server_specific/member_stats.csv',
+                          'r') as stats_file:
+                    contents = stats_file.read()
+                    await ctx.send('note: recorded timezone is PST (UTC -8)')
+                    await ctx.send(f'{contents}')
+                    stats_file.close()
+            else:
+                await ctx.send(f'could not find the stats file!')
+
+    @commands.command()
+    async def m_stats_clear(self, ctx):
+        if not await is_moderator(ctx):
+            return
+        if not await is_mod_commands_channel(ctx):
+            return
+
+        author_id = ctx.author.id
+        if not await is_process_and_user_clear(self.bot, 'm_stats_clear', ctx):
+            return
+
+        self.bot.processes['m_stats_clear'] = author_id
+
+        def check(m):
+            return (
+                    asyncless_is_moderator(m) and
+                    asyncless_is_mod_commands_channel(m) and
+                    m.author.id == author_id and
+                    not m.content.startswith('$m')
+            )
+
+        await ctx.send(f'clear stats file? \'y\' or \'n\'')
+
+        while True:
+
```

Comment:
Personally, I'd delete this blank line. (PEP8 says to generally use blank lines sparingly within functions)